pkg/manager/member/tiflash_scaler_test.go

```diff
@@ -0,0 +1,1727 @@
+						pvcControl.SetDeletePVCError(errors.NewInternalError(fmt.Errorf("API server failed")), 0)
+					},
+					replicas:    5,
+					deleteSlots: sets.NewInt32(),
+				}, {
+					preHandler: func(set *apps.StatefulSet, tc *v1alpha1.TidbCluster, pvcIndexer cache.Indexer, pvcControl *controller.FakePVCControl) {
+						pvcIndexer.Delete(newPVCWithDeleteAnnotaion(set, v1alpha1.TiFlashMemberType, tc.GetName(), 5))
+						pvcControl.SetDeletePVCError(nil, 0)
+					},
+					replicas:    7,
+					deleteSlots: sets.NewInt32(),
+				},
+			},
+		}, {
+			name:           "scale second error with deleteSlots enable asts",
+			enableAsts:     true,
+			oldDeleteSlots: sets.NewInt32(),
+			newDeleteSlots: sets.NewInt32(5, 6),
+			ops: []scaleOp{
+				{
+					preHandler: func(set *apps.StatefulSet, tc *v1alpha1.TidbCluster, pvcIndexer cache.Indexer, pvcControl *controller.FakePVCControl) {
+						pvcIndexer.Add(newPVCWithDeleteAnnotaion(set, v1alpha1.TiFlashMemberType, tc.GetName(), 8))
+						pvcControl.SetDeletePVCError(errors.NewInternalError(fmt.Errorf("API server failed")), 0)
+					},
+					replicas:    6,
+					deleteSlots: sets.NewInt32(5, 6),
+				}, {
+					preHandler: func(set *apps.StatefulSet, tc *v1alpha1.TidbCluster, pvcIndexer cache.Indexer, pvcControl *controller.FakePVCControl) {
+						pvcIndexer.Delete(newPVCWithDeleteAnnotaion(set, v1alpha1.TiFlashMemberType, tc.GetName(), 8))
+						pvcControl.SetDeletePVCError(nil, 0)
+					},
+					replicas:    7,
+					deleteSlots: sets.NewInt32(5, 6),
+				},
+			},
+		}, {
+			name:           "scale first error with deleteSlots enable asts",
+			enableAsts:     true,
+			oldDeleteSlots: sets.NewInt32(),
+			newDeleteSlots: sets.NewInt32(5, 6),
+			ops: []scaleOp{
+				{
+					preHandler: func(set *apps.StatefulSet, tc *v1alpha1.TidbCluster, pvcIndexer cache.Indexer, pvcControl *controller.FakePVCControl) {
+						pvcIndexer.Add(newPVCWithDeleteAnnotaion(set, v1alpha1.TiFlashMemberType, tc.GetName(), 7))
+						pvcControl.SetDeletePVCError(errors.NewInternalError(fmt.Errorf("API server failed")), 0)
+					},
+					replicas:    5,
+					deleteSlots: sets.NewInt32(),
+				}, {
+					preHandler: func(set *apps.StatefulSet, tc *v1alpha1.TidbCluster, pvcIndexer cache.Indexer, pvcControl *controller.FakePVCControl) {
+						pvcIndexer.Delete(newPVCWithDeleteAnnotaion(set, v1alpha1.TiFlashMemberType, tc.GetName(), 7))
+						pvcControl.SetDeletePVCError(nil, 0)
+					},
+					replicas:    7,
+					deleteSlots: sets.NewInt32(5, 6),
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testFn(tt, t)
+		})
+	}
+}
+
+func TestTiFlashScalerScaleIn(t *testing.T) {
+	g := NewGomegaWithT(t)
+	type testcase struct {
+		name          string
+		storeFun      func(tc *v1alpha1.TidbCluster)
+		delStoreErr   bool
+		hasPVC        bool
+		storeIDSynced bool
+		isPodReady    bool
+		hasSynced     bool
+		pvcUpdateErr  bool
```

Comment:
`delStoreErr` and `pvcUpdateErr` always be `false`, and ditto for other `TestXXX` cases